drivers/gpio/gpio_ite_it8xxx2.c

```diff
@@ -0,0 +1,465 @@
+				[gpio_config->gpios_pin]);
+			SET_MASK(*wui_local->reg_addr, wui_local->pin);
+			SET_MASK(*wui_local->clear_addr, wui_local->pin);
+			if (both_tri_en == 1) {
+				SET_MASK(*wui_local->reg_addr, wui_local->pin);
+			}
+		}
+	}
+
+	/* Enable IRQ */
+	ret = irq_connect_dynamic(
+		gpio_config->gpio_irq, 0, gpio_ite_isr, dev, 0);
+	ite_intc_irq_enable(gpio_config->gpio_irq);
+	return ret;
+}
+
+static const struct gpio_driver_api gpio_ite_driver_api = {
+	.pin_configure = gpio_ite_configure,
+	.port_get_raw = gpio_ite_port_get_raw,
+	.port_set_masked_raw = gpio_ite_port_set_masked_raw,
+	.port_set_bits_raw = gpio_ite_port_set_bits_raw,
+	.port_clear_bits_raw = gpio_ite_port_clear_bits_raw,
+	.port_toggle_bits = gpio_ite_port_toggle_bits,
+	.pin_interrupt_configure = gpio_ite_pin_interrupt_configure,
+	.manage_callback = gpio_ite_manage_callback,
+};
+
+static int gpio_ite_init(const struct device *dev)
+{
+	const struct gpio_ite_cfg *gpio_config = DEV_GPIO_CFG(dev);
+	uint8_t regv;
+	unsigned int gpcr_offset = GPCR_OFFSET;
+	uint32_t gpcr_reg;
+	uint32_t gpcr_reg_addr;
+
+	if (gpio_config->reg_addr == GPIO_GPDRM) {
+		gpcr_offset = GPIO_DIR_OUTPUT;
+	}
+
+	gpcr_reg = ((gpio_config->reg_addr & 0xff) - 1) * NUM_IO_MAX
+			+ gpcr_offset;
+	gpcr_reg_addr = gpcr_reg | (gpio_config->reg_addr & 0xffffff00);
+
+	if (gpio_config->port_is_output) {
+		ite_write(gpcr_reg_addr + gpio_config->gpios_pin,
+				1, GPIO_DIR_OUTPUT);
+		if (gpio_config->flag & GPIO_OUTPUT_INIT_HIGH) {
+			set_bit(gpio_config, gpio_config->gpios_pin,
+					GPIO_HIGH);
+		} else {
+			set_bit(gpio_config, gpio_config->gpios_pin,
+					GPIO_LOW);
+		}
+	} else {
+		ite_write(gpcr_reg_addr + gpio_config->gpios_pin, 1,
+				GPIO_DIR_INPUT);
+		regv = ite_read(gpio_config->reg_addr, gpio_config->reg_size);
+	}
+	if (gpio_config->flag & GPIO_INT_ENABLE) {
+		gpio_pin_interrupt_configure(dev,
+					     gpio_config->gpios_pin,
+					     gpio_config->flag);
+	}
+	return 0;
+}
+
+#define GPIO_ITE_DEV_CFG_DATA(n, idx, is_out) \
+static struct gpio_ite_data gpio_ite_data_##n##_##idx; \
+static const struct gpio_ite_cfg gpio_ite_cfg_##n##_##idx = { \
+	.reg_addr = DT_REG_ADDR_BY_IDX(DT_PHANDLE_BY_IDX\
+			(DT_NODELABEL(n), gpios, idx), idx),\
+	.label = DT_LABEL(DT_PHANDLE_BY_IDX(DT_NODELABEL(n), gpios, idx)), \
+	.gpios_pin = DT_PHA_BY_IDX(DT_NODELABEL(n), gpios, idx, pin), \
+	.port_pin_mask = BIT(DT_PHA_BY_IDX(DT_NODELABEL(n), gpios, idx, pin)),\
+	.port_is_output = is_out, \
+	}; \
+DEVICE_AND_API_INIT(gpio_it8xxx2_dev_##n##_##idx, \
+		DT_PROP(DT_NODELABEL(n), label), \
+		gpio_ite_init, \
+		&gpio_ite_data_##n##_##idx, \
+		&gpio_ite_cfg_##n##_##idx, \
+		POST_KERNEL, \
+		CONFIG_KERNEL_INIT_PRIORITY_DEFAULT, \
+		&gpio_ite_driver_api)
+#if DT_NODE_HAS_STATUS(DT_INST(0, gpio_leds), okay)
+GPIO_ITE_DEV_CFG_DATA(room_led, 0, 1);
+#endif /* gpio_leds */
+#if DT_NODE_HAS_STATUS(DT_INST(0, gpio_keys), okay)
+GPIO_ITE_DEV_CFG_DATA(room_sw, 0, 0);
+#endif /* gpio_keys */
```

Comment:
The driver cannot refer to gpio_leds, gpio_keys nodes. This is application specific.

Reply:
It seem's I got them mixed up. I have rewrited this section, I use gpio group as dev init. 
Thanks for your reviewing.